pkg/compare/testdata/ReferenceV2InlineRegex/localWithDiffInFirstLineout.golden

```diff
@@ -0,0 +1,24 @@
+**********************************
+
+Cluster CR: v1_ConfigMap_kubernetes-dashboard_kubernetes-dashboard-settings
+Reference File: cm-regex-with-diff-in-first-line.yaml
+Diff Output: diff -u -N TEMP/v1_configmap_kubernetes-dashboard_kubernetes-dashboard-settings TEMP/v1_configmap_kubernetes-dashboard_kubernetes-dashboard-settings
+--- TEMP/v1_configmap_kubernetes-dashboard_kubernetes-dashboard-settings	DATE
++++ TEMP/v1_configmap_kubernetes-dashboard_kubernetes-dashboard-settings	DATE
+@@ -7,5 +7,5 @@
+   namespace: kubernetes-dashboard
+ spec:
+   bigTextBlock: |-
+-    This is a big text block with some static content, like this line
+-    It also has a place where (?<username>[a-z0-9]+) would put in their own name. (?<username>[a-z0-9]+) would put in their own name.
++    This is a big text block with some static content, like this line.
++    It also has a place where exampleuser would put in their own name. exampleuser would put in their own name.
```

Comment:
I can see why the 1st line is different; the missing period at the end.  But why is this 2nd line showing up as different?

Reply:
that is why i said the test probably does not have enough value. because in this version the function only check the if the expression matches the regex   and if it does not the value that will be in template is the regex if it does match then it will be set to the value in the cluster cr resulting in no diffs.
for future work we can upgrade the mechanism to return no the regex but divvide it in to lines and run the regex separately per each line and then return a blob that will partially contain the regex and partially the clusters data.
this feature needs a little bit of thinking because what happens if a line is missing and other scenarios

Reply:
Ah, I see!  Perhaps I was expecting we would do a line-by-line match, but I understand why we're not yet.